test/osb_and_plugin_test/osb_test/instance_sharing_test.go

```diff
@@ -0,0 +1,394 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package osb_test
```

Comment:
add a test to confirm what happens when we try to unshare a shared instance that has references

Reply:
added:
fails unsharing an instance with references
fails re-sharing a shared instance
fails sharing an instance with invalid share request

Reply:
adding more tests for shared instance without references

Reply:
@alexwo - I added the new tests, can you review them? 
test/service_instance_and_binding_test/service_instance_test/service_instance_test.go
describe: patch
context: shared instance with references
and
context: shared instance without references